modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java

```diff
@@ -135,4 +145,43 @@ private String unwrapChildKey(String key)
     {
         return key.replaceAll(CHILD_ELEMENT_KEY_PREFIX, "").replaceAll(CHILD_ELEMENT_KEY_SUFFIX, "");
     }
+
+    private void checkParameterGroupExclusivenessForModel(EnrichableModel model, ResolverSet resolverSet) throws ConfigurationException
+    {
+        Optional<ParameterGroupModelProperty> parameterGroupModelProperty = model.getModelProperty(ParameterGroupModelProperty.class);
+        if (parameterGroupModelProperty.isPresent() && parameterGroupModelProperty.get().hasExclusion())
+        {
+            checkParameterGroupExclusiveness(parameterGroupModelProperty.get(), resolverSet);
+        }
+    }
+
+    private void checkParameterGroupExclusiveness(ParameterGroupModelProperty parameterGroupModelProperty, ResolverSet resolverSet) throws ConfigurationException
+    {
+        Set<String> resolverKeys = resolverSet.getResolvers().keySet();
+        for (ParameterGroup group : parameterGroupModelProperty.getGroups())
+        {
+            Set<String> parameterGroupFields = group.getParameters().stream().map(f -> f.getName()).collect(Collectors.toSet());
+
+            Optional<String> foundParameter = Optional.empty();
+            for (String parameterName : resolverKeys)
+            {
+                if (parameterGroupFields.contains(parameterName))
+                {
+                    if (!foundParameter.isPresent())
+                    {
+                        foundParameter = Optional.of(parameterName);
+                    }
+                    else
+                    {
+                        throw new ConfigurationException(createStaticMessage(format("Parameters '%s' and '%s' from class '%s' are exclusive between each other and cannot be both present at the same time", parameterName, foundParameter.get(), group.getType().getName())));
```

Comment:
we should communicate all the parameters that are conflicting, not just the first two found (we can have an exclusion of three with all of them configured).
Something like: 
``` 
parametersFromGroup = group.getParameters().stream()
                            .filter(f -> resolverKeys.contains(f.getName()))  
                            .collect(toSet());  //all the parameters from the group that have a value assigned

if (parametersFromGroup.size() > 1){
    throw new ConfigurationException("something like: Only one of this parameters is allowed [<parametersFromGroup>]" );
}

if (group.atLeastOneIsRequired() && parametersFromGroup.isEmpty())
{
    throw new ConfigurationException(...);
}
```